SofaKernel/modules/SofaBaseMechanics/DiagonalMass.inl

```diff
@@ -224,15 +313,37 @@ void DiagonalMass<DataTypes,MassType>::DMassPointHandler::ApplyTopologyChange(co
     const sofa::helper::vector< sofa::helper::vector< unsigned int > >& ancestors = e->ancestorsList;
     const sofa::helper::vector< sofa::helper::vector< double > >& coefs = e->coefs;
 
-    applyTriangleCreation(triangleAdded,elems,ancestors,coefs);
+    if(dm->triangleGeo)
```

Comment:
To avoid all those Geometry component check, I would put that test in the init method when the callback are created. Thus if for example you are using the mass on the Triangulation, TetraGeo and hexaGeo would be nullptr and so you don't create the call back for this level of topology.